Calypso style search and subnav search

joshuatf opened the pull request:
This PR styles the search box to look like Calypso's and also moves it into the subnav if one exists on the page.

Fixes #151 

#### Screenshots
<img width="795" alt="screen shot 2018-11-12 at 2 31 37 pm" src="https://user-images.githubusercontent.com/10561050/48343532-99de1800-e6ad-11e8-9083-eaafafc3c92b.png">
<img width="786" alt="screen shot 2018-11-12 at 2 31 44 pm" src="https://user-images.githubusercontent.com/10561050/48343534-99de1800-e6ad-11e8-87fc-6b1d269c0caa.png">
<img width="802" alt="screen shot 2018-11-12 at 2 31 47 pm" src="https://user-images.githubusercontent.com/10561050/48343535-99de1800-e6ad-11e8-9c4d-555d605e622e.png">

#### Testing
1.  Visit pages with search boxes both with a subnav and without.  E.g., `/wp-admin/edit-tags.php?taxonomy=product_cat&post_type=product` and `/wp-admin/edit.php?post_type=product`
2.  Check that styling and functionality work for both types and match Calypso's - https://wordpress.com/store/products/

##### Some things to test for
* Search icon (magnifying glass) should open the input on products and focus the input.
* Close icon (x) should only be visible in subnav search bar and should close menu.
* Search should still work on both

#### Design
The Calypso style search runs on type, while the WP forms only run on submit.  I've kept the submit buttons inside the form for the time being, but should we change this?  /cc @josemarques 

We could also:
* Auto-submit when the user finishes typing
* Use a different icon for search?

josemarques commented:
> The Calypso style search runs on type, while the WP forms only run on submit. I've kept the submit buttons inside the form for the time being, but should we change this? /cc @josemarques

Is it possible to submit on enter and hide the button?

joshuatf commented:
@josemarques Sure!  Hid buttons and added a function to allow search on 'enter.'

@justinshreve Good idea on the auto-expand if a query is set.  Added that in.  Also disabled the autocomplete since it looks like this is turned off in Calypso.
